cmd/contour/serve.go

```diff
@@ -241,8 +241,14 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 	informerSyncList.InformOnResources(clusterInformerFactory, dynamicHandler, k8s.DefaultResources()...)
 
 	if ctx.UseExperimentalServiceAPITypes {
-		informerSyncList.InformOnResources(clusterInformerFactory,
-			dynamicHandler, k8s.ServiceAPIResources()...)
+		for _, s := range k8s.ServiceAPIResources() {
+			// Check if the resource exists in the API server before setting up the informer.
+			if !clients.ResourceExists(s) {
+				log.WithField("InformOnResources", "ExperimentalServiceAPITypes").Warnf("resource %v not found in api server", s.String())
+				continue
+			}
+			informerSyncList.InformOnResources(clusterInformerFactory, dynamicHandler, s)
```

Comment:
Does it make sense to try to keep going if some resources are missing? I think that it would be better (or at least more predictable) to make it all or nothing.

I suggest something like the following:
```Go
apiResources, err := clients.ServerResources()
if err != nil {
        log.Fatal(err) // Not being able to query API server resources is totally broken.
}

if ctx.UseExperimentalServiceAPITypes) {
        // ".Contains()" would return true if all the arguments are present in the set of resources.
        if apiResources.Contains(k8s.ServiceAPIResources()...) {
                informerSyncList.InformOnResources(...)
        } else {
                log.Warnf("APIs not present)
        }
}
```

This approach lets the caller determine how to handle any errors (I think we mostly would want to fail hard on this). It also avoids any awkward questions about revalidating the cache copy in the client since we can also push that onto the caller.

WDYT?

Reply:
Sure thing, I pushed an update.

Reply:
I approved, but I still think that we should avoid caching the resources in the client.